spec.md

```diff
@@ -0,0 +1,238 @@
+
+> **Note:** The `config.mediaType` of `application/vnd.oci.image.config.v1+json` is reserved for artifacts intended to be instanced by [docker][docker] and [containerd][containerd].  
+*Each artifact MUST have its own unique type.*
+
+ The following `mediaType` format is used to differentiate the type of artifact:
+
+`application/vnd.`[org|company]`.`[objectType]`.`[optionalSubType]`.config.`[version]`+json`
+
+- **`org|company`** - represents an open source foundation (`oci`, `cncf`) or a company (`microsoft`, `ibm`).
+- **`objectType`** - a value representing the short name of the type. The combination of the org|company and object type should provide human identification and uniqueness for computers indexing.
+- **`optionalSubType`** - provides additional extensibility of an `objectType`
+- **`version`** - provides artifact authors the ability to revision their schemas and formatting, enabling tools to identify which format they will process.
+
+## Defining Supported Layer Types
+
+Artifacts are intended to have content. The content of an artifact is represented through a collection of blobs as provided by the [distribution spec][distribution-spec]. How the blobs are re-constituted, and whether the blobs are ordinal layers is a decision of the artifact author.
+
+As an example, [OCI Images][image-layer] are represented through an ordinal collection of compressed files. Each blob represents a layer. Each layer overlays the previous layer.
+
+Other artifacts may be represented as a single file, such as config artifact representing a deployment specification. Other artifacts may include a config blob, and a collection of binaries compressed as another blob. By separating the blob, the artifact author can benefit from layer de-duplication and concurrent downloading of each blob.
+
+### Layer File Format
+
+Layers are persisted as blobs within registries. The blobs can be a single file or a collection of files. The persistance format is up to the artifact author. They may choose to persist individual files with their native or custom format, such as `.config`, `.json`, `.bin`, `.sif`, and compress them with various formats such as `.gzip`.
+
+Large files can benefit from compression when being transferred across the network. However, decompression takes time and compute cycles. For smaller, individual files, the decompression may take longer than downloading the file in its original format.
+
+### Layer Versioning
+
+Layers MUST be versioned to future proof new enhancements that may evolve. How the artifact tooling processes forward and backward compatibility is up to the artifact and tooling authors.
+
+### Defining Layer Types
+
+Artifact layers are expected to be unique, per artifact author. A layer that represents an OCI Image is not expected to be shared with a Helm Chart. To differentiate layers, artifact authors SHOULD create unique artifact layer types.
```

Comment:
What does it mean "expected to be unique"? New types should only be defined if it is actually defining something new. If the layer is just a tar without any special processing, then just a standard tar type should be used. If the layer uses the OCI layer definitions, it should just use that. In reality the bottommost layer in an OCI image could be defined as "application/tar" (this wouldn't be supported by most clients), but there is no need to define something new unless it is actually unique.

Reply:
Also, content-addressable digests give solid uniqueness guarantees regardless of the nominal media type.

Reply:
I don't see why a security scanner would care about the consuming context when processing a tar file.  Can you float a hypothetical example?

Also, the image spec already requires manifest consumers to be able to support [some media types][1] for layers.  And it [requires the layers to be ordered][2].  Is the motivation for trying to overload the existing manifest type "registries already support it"?  It seems like it might be easier to leave the type alone and relax the distribution spec to allow folks to tag arbitrary media types where the distribution spec currently assumes they'll be [tagging manifests or index files][3] (although it currently says nothing explicit about acceptable media types for that PUT and just handwaves around the `MANIFEST_INVALID` conditions).

[1]: https://github.com/opencontainers/image-spec/blame/v1.0.1/manifest.md#L55
[2]: https://github.com/opencontainers/image-spec/blame/v1.0.1/manifest.md#L46
[3]: https://github.com/opencontainers/distribution-spec/blob/v1.0.0-rc0/spec.md#put-manifest

Reply:
As security scanners look at the contents, the context adds more information to the files for how they're executed or vulnerable. For instance, the image-spec does specify ordering, with whiteout. 

The image-spec is one type of artifact. Helm, Singularity, CNAB, ... are other equal types of artifacts that can define their own rules. blob ordering, representing layers. Other artifact do not need to specify this concept. 

A scanner may care about the configuration, that specifies parameters, or provides a value that is considered unsafe. 

The pattern here is fitting between distribution and image-spec.

Reply:
re-reading the spec and the feedback, I believe I've cleaned up the confusion. Please reopen if the update isn't clear. Or, any other suggestions.